src/openassetio-core/src/hostApi/Manager.cpp

```diff
@@ -23,6 +23,44 @@
 namespace openassetio {
 inline namespace OPENASSETIO_CORE_ABI_VERSION {
 namespace {
+
+/**
+ * Validate the supplied ManagerInterface supports all required
+ * capabilities, or throw a ConfigurationException.
+ *
+ * These capabilities are required at runtime, but not enforced at
+ * compile time, as multi-language implementations may only provide the
+ * implementation in one of the component languages.
+ */
+void verifyRequiredCapabilities(const managerApi::ManagerInterfacePtr &interface) {
+  using managerApi::ManagerInterface;
+  using CapabilityVector = std::vector<ManagerInterface::Capability>;
+
+  static const CapabilityVector kRequiredCapabilities = {
```

Comment:
[consider][minor] Using `std::array` as the type here might let you get away with declaring this `constexpr` in c++17.

Reply:
Nice, thanks. Updated [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/1156/commits/90a0e40cdf4fd45e4a7143a26a2ae404b4fc7e67)